### PR TITLE
New setup

### PR DESCRIPTION
**Pull request by onefastsnail**
## Changes

No real code changes or code improvements but rather focusing on developer experience and helping the overall maintenance of the project by creating a simpler easy stack to develop with. Allowing others to contribute more easily, hopefully 😅 .

This sat on machine for quite sometime, only until recent activity i then was reminded i had this, so here we go... 😄

The Prettier change makes this PR slightly harder to look at, should have done this in a new PR, my bad.

- [x] Using Heroku for a free SSL and CI pipeline.
- [x] Using a Heroku Worker to detach the cron job we had running off the main web process
- [x] Making the code Prettier
- [x] Added a Github code owners file to automagically assign me and you to PR reviews.
- [x] Added Github issue and pull request templates to help structure incoming contributions
- [x] Allow easier local development, this a more configurable setup via some env vars
- [x] Removed OG Bower dep
- [x] Moved to Webpack
- [x] A small UI cleanup
- [x] A small code clean up, mostly removing unused code
- [ ] Add a linter
- [ ] Moved to Typescript? May be easier with a full rewrite. Wouldn't take long.
- [ ] Add a Papertrail transport layer Winston? 
- [ ] Update to Node 12x

The unchecked items i figured would be best in a separate PRs and not further extend this chaotic PR.

I have a working demo at https://strava2kilometrikisa.herokuapp.com with the pipelines implemented.

The flow would go as follows: 
1. Feature branch development
1. Pull request into master
1. Once merged into master, a UAT version is automatically built and deployed to https://strava2kilometrikisa-uat.herokuapp.com. 
1. Then once we are happy with it we can promote to the live env. Up next we could add the Review Apps integration in Heroku to allow easier PR reviews. But i figured for now the above will suffice.

I originally tried to detach the sync to a scheduled event, but our sync service requires X time per registered user whilst syncing, so such a call could take minutes and only grows with users, where the worker feels more capable in this use case. However the daily restart of Heroku dynos may occur during a sync, for now i would suggest to run the sync twice a day to at least get a successful sync once a day. The API rate limits "should" be fine 😅 . 

Both demo versions use a test MongoDB so feel free to spam and test. Sync is 4am on both environments.

## Moving forward

1. To point to the current DNS to our friends at Heroku
1. We are currently using a free tier on Heroku, so the dyno will sleep, we may want to bump to the Hobby tier to alleviate this issue, or at least create a dedicated strava2kilometrikisa, as its currently under my account.
1. And Enjoy :).

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# see: http://EditorConfig.org
```
Comment (hugovk) [inline .editorconfig]:
```suggestion
# see: https://EditorConfig.org
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
```
Comment (hugovk) [inline .github/CONTRIBUTING.md]:
```suggestion
## We develop with GitHub
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+We use github to host code, to track issues and feature requests, as well as accept pull requests which trigger deployments.
```
Comment (hugovk) [inline .github/CONTRIBUTING.md]:
```suggestion
We use GitHub to host code, to track issues and feature requests, as well as accept pull requests which trigger deployments.
```

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
```
Comment (hugovk) [inline .github/CONTRIBUTING.md]:
```suggestion
## We use [GitHub Flow](https://guides.github.com/introduction/flow/index.html), so all code changes happen through pull requests
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+Pull requests are the best way to propose changes to the codebase (we use [Github Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
+
+1. Fork the repo and create your branch from `master`.
+1. ~~If you've added code that should be tested, add tests.~~ WIP
+1. Make sure your code lints.
+1. ~~Ensure the test suite passes~~. WIP
+1. [Create a pull request](https://github.com/jaamo/strava2kilometrikisa/compare)!
+
+## Report bugs using Github's [issues](https://github.com/briandk/transcriptase-atom/issues)
```
Comment (hugovk) [inline .github/CONTRIBUTING.md]:
```suggestion
## Report bugs using GitHub's [issues](https://github.com/briandk/transcriptase-atom/issues)
```

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+## Expected Behavior
```
Comment (hugovk) [inline .github/ISSUE_TEMPLATE.md]:
```suggestion
## Expected behavior
```

File: .github/ISSUE_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+<!--- Describe what should happen -->
+
+## Current Behavior
```
Comment (hugovk) [inline .github/ISSUE_TEMPLATE.md]:
```suggestion
## Current behavior
```

File: .github/ISSUE_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+<!--- Describe what happens instead of the expected behavior -->
+
+## Steps to Reproduce
```
Comment (hugovk) [inline .github/ISSUE_TEMPLATE.md]:
```suggestion
## Steps to reproduce
```

File: .github/ISSUE_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+<!--- Provide a link to the environment, or an unambiguous set of steps to help reproduce the issue -->
+1.
+1.
+1.
+1.
+
+## Environment
+<!--- Some details of the environment to help reproduce the issue in the same environment -->
+
+## Possible Solution
```
Comment (hugovk) [inline .github/ISSUE_TEMPLATE.md]:
```suggestion
## Possible solution
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+## Description of change
+<!--- Describe your changes. Why are these changes required? What problem does it solve? -->
+
+## Related Issue
```
Comment (hugovk) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
## Related issue
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+<!--- Please link to the issue here, ideally through your commit messages -->
+
+## How Has This Been Tested?
```
Comment (hugovk) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
## How has this been tested?
```

**Comment by hugovk** [comment]
Well, the working demo worked to manually sync my existing activities for the last week! 👍

**Comment by onefastsnail** [comment]
Awesome! And thanks for the feedback @hugovk, will get onto these changes tomorrow 😄.

File: app/models/UserModel.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  notifiedByEmail: { type: Boolean },
+});
+
+UserSchema.methods.updateToken = function() {
+  var d = new Date();
+  if (d > this.tokenExpire) {
+    const payload = strava.oauth.refreshToken(this.refreshToken);
+    this.stravaToken = payload.access_token;
+    this.tokenExpire = payload.expires_at * 1000;
+    this.refreshToken = payload.refresh_token;
```
Comment (Mattti0) [inline app/models/UserModel.js]: This should be like in PR #32

Comment (onefastsnail) [reply app/models/UserModel.js]: Thanks for the pointer and PR @Mattti0 ! 😍 

I've now merged #32 into this PR.

**Comment by onefastsnail** [comment]
Currently putting a fix onto #32 to query the Strava APIs only once the `updateToken` method has resolved, ie with a token. BRB 😄

**Comment by onefastsnail** [comment]
Any further input on this @jaamo, @hugovk and @Mattti0? 😄

**Comment by onefastsnail** [comment]
> I've started using the demo site each day to manually sync.

Awesome!

> I guess there's no cron on the demo site, either that some missing settings or a bug in updating each day.

Yeh the worker was disabled as it would eat away at my Heroku allowance 😅. It is now running and will run a sync at 22:00 tonight if you want to test this @hugovk?

**Comment by hugovk** [comment]
> It is now running and will run a sync at 22:00 tonight if you want to test this @hugovk?

Yes, will report back!

**Comment by onefastsnail** [comment]
Turned out I created an issue regarding a lack of persisting the user token to the DB which now seems kind of obvious 😅 and the usage of the newly generated token once the promise resolves, all of which have been fixed and deployed to both UAT and LIVE environments of the demo site.

The worker did run at 22:00 as intended, looped through the users in the test DB, all of which failed to query the Strava API due to invalid tokens. So debugging locally and running a local worker with this fix, all worked and sync all test DB users.

**Comment by Mattti0** [comment]
Seems like it has automatically synced also my data to Kilometrikisa! Last time I logged to site was maybe about week ago. Thanks!

**Comment by hugovk** [comment]
👍 Yesterday's 18:00 ride has been synced fine to Kilometrikisa.

**Comment by onefastsnail** [comment]
Thanks for the feedback @Mattti0 and @hugovk.

I just ran the worker on the new production demo environment and it synced all the DB users activities, and testing my Kilometrikisa data, all looks 👌 this morning. 

And @Mattti0's token refresh fix worked perfect to update the short-lived access tokens.

If all ok with everyone, I will merge this to master and speak with @jaamo to update the DNS.

Thanks for all your help! 😄

**Comment by hugovk** [comment]
The new setup has been working well so far!

I have a gap for the 1-22 January, would it be possible to set up a demo site to sync those dates? Or do you have instructions on how to set it up locally to do so?

Thanks!

**Pull request merged**